File: TodoApi/Models/StorageAreas/Domain/StorageArea.cs
using System.Collections.Generic;
using FrameworkDDD.Common;

namespace TodoApi.Models.StorageAreas
{
    public enum StorageAreaType
    {
        Yard,
        Warehouse
    }

    public class StorageArea : IAggregateRoot
    {
        public int Id { get; set; } // Unique identifier
        public StorageAreaType Type { get; set; }
    public string Location { get; set; } = string.Empty; // Location within the port
        public int MaxCapacityTEU { get; set; } // Maximum capacity in TEUs
        public int CurrentOccupancyTEU { get; set; } // Current occupancy in TEUs

        // By default, serves all docks. If not, specify dock IDs.
        public List<int> ServedDockIds { get; set; } = new List<int>();

        // Complementary info: distances to docks (dockId -> distance in meters)
        public Dictionary<int, double> DockDistances { get; set; } = new Dictionary<int, double>();
    }
}

File: frontend/src/app/models/storage-area.ts
export interface StorageAreaDTO {
  id: number;
  type?: string | null;
  location?: string | null;
  maxCapacityTEU: number;
  currentOccupancyTEU: number;
  servedDockIds?: number[];
  dockDistances?: Record<number, number>;
}

export interface CreateStorageAreaDTO {
  type: string;
  location: string;
  maxCapacityTEU: number;
  currentOccupancyTEU: number;
  servedDockIds?: number[];
  dockDistances?: Record<number, number>;
}

export interface UpdateStorageAreaDTO extends CreateStorageAreaDTO {
  id: number;
}
